pycaret/internal/plots/time_series.py

```diff
@@ -1036,3 +1047,192 @@ def plot_time_series_decomposition(
+    lags_list = data_kwargs.get("lags_list", None)
+
+    title_name = f"Difference Plot"
+    data_name = data.name if model_name is None else f"'{model_name}' Residuals"
+
+    if model_name is None:
+        title = f"{title_name}" if data.name is None else f"{title_name} | {data_name}"
+    else:
+        title = f"{title_name} | {data_name}"
+
+    diff_list, name_list = get_diffs(
+        data=data, order_list=order_list, lags_list=lags_list
+    )
+
+    if len(diff_list) == 0:
+        # Issue with reconciliation of orders and diffs
+        return fig, return_data_dict
+
+    diff_list = [data] + diff_list
+    name_list = ["Actual" if model_name is None else "Residuals"] + name_list
+
+    rows = len(diff_list)
+    fig = make_subplots(rows=rows, cols=1, row_titles=name_list, shared_xaxes=True,)
+
+    for i, subplot_data in enumerate(diff_list):
+        x = (
+            subplot_data.index.to_timestamp()
+            if isinstance(data.index, pd.PeriodIndex)
+            else data.index
+        )
+        fig.add_trace(
+            go.Scatter(
+                x=x,
+                y=subplot_data.values,
+                line=dict(color="#1f77b4", width=2),
+                mode="lines+markers",
+                name=name_list[i],
+                marker=dict(size=5,),
+            ),
+            row=i + 1,
+            col=1,
+        )
+
+    fig.update_layout(title=title)
+    fig.update_layout(showlegend=False)
+    fig_template = fig_kwargs.get("fig_template", "ggplot2")
+    fig.update_layout(template=fig_template)
+
+    fig_size = fig_kwargs.get("fig_size", [800, 200 * rows])
+    if fig_size is not None:
+        fig.update_layout(
+            autosize=False, width=fig_size[0], height=fig_size[1],
+        )
+
+    return_data_dict = {
+        "data": data,
+        "diff_list": diff_list,
+        "name_list": name_list,
+    }
+
+    return fig, return_data_dict
+
+
+##########################
+#### Helper Functions ####
+##########################
+
+
+def __reconcile_order_and_lags(
+    order_list: Optional[List[Any]] = None, lags_list: Optional[List[Any]] = None
+) -> Tuple[List[int], List[str]]:
+    """Reconciles the differences to lags and returns the names
+    If order_list is provided, it is converted to lags_list.
+    If lags_list is provided, it is uses as is.
+    If none are provided, assumes order = [1]
+    If both are provided, returns empty lists
+
+    Parameters
+    ----------
+    order_list : Optional[List[Any]], optional
+        order of the differences, by default None
+    lags_list : Optional[List[Any]], optional
+        lags of the differences, by default None
+
+    Returns
+    -------
+    Tuple[List[int], List[str]]
+        (1) Reconciled lags_list AND
+        (2) Names corresponding to the difference lags
+    """
+
+    return_lags = []
+    return_names = []
+
+    if order_list is not None and lags_list is not None:
+        print(
```

Comment:
Raise an exception? if that's not acceptable, then please use `warnings.warn` instead